examples/navier-stokes/advection.h

```diff
@@ -20,11 +20,52 @@
 #ifndef advection_h
 #define advection_h
 
-#include <math.h>
+#ifndef __CUDACC__
+#  include <math.h>
+#endif
+
+#ifndef setup_context_struct
+#define setup_context_struct
+typedef struct SetupContext_ *SetupContext;
+struct SetupContext_ {
+  CeedScalar theta0;
+  CeedScalar thetaC;
+  CeedScalar P0;
+  CeedScalar N;
+  CeedScalar cv;
+  CeedScalar cp;
+  CeedScalar Rd;
+  CeedScalar g;
+  CeedScalar rc;
+  CeedScalar lx;
+  CeedScalar ly;
+  CeedScalar lz;
+  CeedScalar periodicity0;
+  CeedScalar periodicity1;
+  CeedScalar periodicity2;
+  CeedScalar center[3];
+  CeedScalar dc_axis[3];
+  CeedScalar time;
+};
+#endif
+
+#ifndef advection_context_struct
+#define advection_context_struct
+typedef struct AdvectionContext_ *AdvectionContext;
+struct AdvectionContext_ {
+  CeedScalar CtauS;
+  CeedScalar strong_form;
+  int stabilization; // See StabilizationType: 0=none, 1=SU, 2=SUPG
+};
+#endif
 
-static int Exact_Advection(CeedInt dim, CeedScalar time, const CeedScalar X[],
-                           CeedInt Nf, CeedScalar q[], void *ctx) {
-  const SetupContext context = ctx;
+// *****************************************************************************
+// TODO: Annotate this function
+// *****************************************************************************
+static inline int Exact_Advection(CeedInt dim, CeedScalar time,
+                                  const CeedScalar X[], CeedInt Nf,
+                                  CeedScalar q[], void *ctx) {
```

Comment:
The VLA stuff works fine now. But. These helper functions are causing NVRTC to choke
```
+default_program(124): error: A function without execution space annotations (__host__/__device__/__global__) is considered a host function, and host functions are not allowed in JIT mode. Consider using -default-device flag to process unannotated functions as __device__ functions in JIT mode
```
@jedbrown, thoughts on what we want to do here?
I think supporting helper functions for QFunctions fosters more maintainable code. I'm hesitant to go macro crazy, but do we need something like `CEED_QFUNCTION()` for helper functions at the library level? We could also put it here in the QFunction source if these will be uncommon, but that breaks the abstraction that QFunction source is backend agnostic.

Reply:
If we can set default device, I think that would be ideal, otherwise the macro. These helpers are important.

Reply:
You can also use the `QFunction` macro I think, which might be cleaner.

Reply:
Yeah, though it'll also create an unused `_loc`.

Reply:
We could consider adding some kind of similar `CEED_FUNCTION` that does not create the `_loc`.

Reply:
I guess `-default-device` flag is also okay since it only concerns `nvrtc`.

Reply:
Yeah, that's the fallback plan. I don't see a reason to do that if we can make `__device__` implicit during JIT. (I don't see a downside to that. I don't think the JIT scope ever needs to include host functions.)

Reply:
I added `-default-device` as a flag.